### ISSUE TITLE
Add currentdir layout renderer

### ISSUE THREAD

**Issue by SanderH** [NLog/NLog#2190]
**Type** (choose one): Feature request

**NLog version**: 4.2.3

**Platform**: .Net 4.5 

**Current NLog config** (xml or C#, if relevant)

```xml
<nlog>
  <targets>
    <target type="File" filename="${currentdir}\Logs\mylogs.log">
  </targets>
  <rules>
  </rules>
</nlog>

```

In previous versions of NLog (2.x) we could use filename="Logs\mylogs.log" in the File target to have the logs relative to the working folder. I think since the default path is ${basedir}, the logs are attempted to be written to the application folder, to which users normally don't have access.

We're using NLog in a commandline tool to import data into our system, we want the logs to be relative to the data the logs belong to. So in this case entering a log location in the config to a fixed path doesn't work.

As a workaround to get the old functionality we have created a simple layout renderer currentdir (similar to basedir) which is just returns Directory.GetCurrentDirectory() like [CurrentDirLayoutRenderer.zip](https://github.com/NLog/NLog/files/1115063/CurrentDirLayoutRenderer.zip) which works properly.
I'm surely not the only one to need this feature, so adding this to the library for others to use will benefit all users of NLog.

**Comment by 304NotModified** [comment]
You can do for now:

```c#
//register $currentdir}
LayoutRenderer.Register("currentdir", (logEvent) =>Directory.GetCurrentDirectory());
```

**Comment by Schobers** [comment]
@SanderH No you're not the only one. To be honest I was a bit surprised this wasn't implemented yet (I searched for a while until I ran into this issue actually).

To put it differently, I'd expect (not regarding backwards compatibility, assumptions on current implementation etc.) this to be the default directory if a relative path was supplied. But as a workaround the ${currentdir} would be nice!

**Comment by UgurAldanmaz** [comment]
Do we have a plan for this @304NotModified?

**Comment by UgurAldanmaz** [comment]
@304NotModified, if we want to implement it, I would like to do.

**Comment by 304NotModified** [comment]
please do :) Would be very nice to have :)

**Comment by snakefoot** [comment]
@304NotModified Resolved with #2491
